### PR TITLE
Sz dscs 2425

### PR DESCRIPTION
**Pull request by shuangzhou**
This request logs and reports records with data too big for amberdb fields in order to help with locating and amended these values in job processing and work records creation.

### CODE REVIEW

**Comment by shuangzhou** [comment]
@scoen @jrixon could you please review? Thanks.

**Comment by jrixon** [comment]
I think this could result in some misleading error messages in the logs. It's not certain that the UnableToExecuteStatementException exception is caused by values being too long - the exception could have been caused by something else entirely (table locking problems, for example).

**Comment by shuangzhou** [comment]
@jrixon I can also check for the wrapped exception and verify the message contains Data too long for column?

**Comment by scoen** [comment]
i think also, we probably don't want the entire too-big entry going into the logs. Maybe just the start of it so it can be identified (maybe include id of work/copy)? Very long lines in the logs makes some shells/viewers go a bit funny when trying to read them.

**Comment by shuangzhou** [comment]
@scoen I'll update to have the first 500 characters.

**Comment by nlabuilder** [comment]
Build finished.

File: src/main/java/amberdb/graph/AmberGraph.java [hunk 1]
```diff
@@ -358,6 +371,35 @@ public Long suspend() {
...
+        }
+        log.error("Failed to suspend due to properties in session: {}", sessId);
+        if (recordToLog != null) {
+            for (String field : recordToLog.keySet()) {
+                log.error("Values of largest record in session {} : {} : {}", sessId, field, recordToLog.get(field));
+            }
+        }
+        return (new ObjectMapper()).writeValueAsString(recordToLog);
+    }
+
```
Comment (scoen) [inline src/main/java/amberdb/graph/AmberGraph.java]: Not sure the linked maps really add much value to the routine. Can you do something like: 

```
// loop through all values
int indexOfLargest = -1;
int largestSize = 0;
for (int i = 0; i < p.value.size; i++) {
    if (p.value.get(i).length > largestSize) {
        indexOfLargest = i;
        largestSize = p.value.get(i).length;
    }
    // now use indexOfLargest as index in the other AmberPropertyBatch arrays to extract name etc as required
    // eg: long idOfLargest = p.id.get(indexOfLargest);
}

```

Comment (scoen) [inline src/main/java/amberdb/graph/AmberGraph.java]: sorry , pull the final comments out of the loop
```// loop through all values
int indexOfLargest = -1;
int largestSize = 0;
for (int i = 0; i < p.value.size; i++) {
    if (p.value.get(i).length > largestSize) {
        indexOfLargest = i;
        largestSize = p.value.get(i).length;
    }
}
// now use indexOfLargest as index in the other AmberPropertyBatch arrays to extract name etc as required
// eg: long idOfLargest = p.id.get(indexOfLargest);

**Comment by nlabuilder** [comment]
Build finished.

**Comment by nlabuilder** [comment]
Build finished.

**Comment by nlabuilder** [comment]
Build finished.

**Comment by nlabuilder** [comment]
Build finished.

**Comment by nlabuilder** [comment]
Build finished.

**Comment by scoen** [comment]
FYI interested parties :-) Swaps the order that sessions are suspended in, to be flat tables before old graph tables. This is so now, when we try to suspend a value that's too big for a given column, the exception that gets thrown tells us which column has the problem rather than just saying some property value is too big.

**Pull request merged**